atd-vze/src/views/Crashes/CrashChangeLog.js

```diff
@@ -0,0 +1,216 @@
+import React, { Component } from "react";
+
+import {
+    Badge,
+    Button,
+    Card,
+    CardBody,
+    CardHeader,
+    Col,
+    Table,
+    Row,
+    Modal, ModalHeader, ModalBody, ModalFooter
+} from "reactstrap";
+
+
+import {Link} from "react-router-dom";
+
+class CrashChangeLog extends Component {
+    constructor(props) {
+        super(props);
+
+        this.closeModal = this.closeModal.bind(this);
+        this.showModal = this.showModal.bind(this);
+
+        this.compare = this.compare.bind(this);
+
+        this.state = {
+            modal: false,
+            modalBody: null,
+            dataFrom: null,
+            dataTo: null,
+            data: this.props.data
+        };
+    }
+
+    /**
+     * Converts a PostgreSQL Timestamp string into a human readable date-time
+     * @param {String} the timestamp as provided by postgres
+     */
+    timeConverter(UNIX_timestamp){
+        let a = new Date(`${UNIX_timestamp}`.replace(' ', 'T'));
+        let months = ['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec'];
+        return a.getDate() + ' ' + months[a.getMonth()] + ' ' + a.getFullYear() + ' ' + a.getHours() + ':' + a.getMinutes();
+    }
+
+    /**
+     * Shows the modal box by changing the state, does not alter the body.
+     */
+    showModal() {
+        this.setState({
+            modal: true
+        });
+    }
+
+    /**
+     * Closes the modal box by changing the state and emptying the modal body.
+     */
+    closeModal() {
+        this.setState({
+            modal: false,
+            modalBody: null
+        });
+    }
+
+    /**
+     * Compares the archived json object against the current data object as populated via props.
+     * @param {Object} record's json
+     */
+    compare(record) {
+        // Holds only the keys
+        let diff = [];
+
+        // Holds the html for our modal box
+        let modalBody = null;
+
+        // Iterate through our record's key & values
+        for (let [key, value] of Object.entries(this.props.data.atd_txdot_crashes[0])) {
+            // Let's get rid of typename
+            if(key === "__typename") continue;
+
+            try {
+                // Gather the archived value for the current field
+                let archivedRecordValue = record.record_json[key];
+
+                // If the value is different from the current value, then put it in the diff array.
+                if(archivedRecordValue !== value) {
+                    diff.push({
+                        'original_record_key': key,
+                        'original_record_value': value,
+                        'archived_record_value': archivedRecordValue
+                    });
+                }
+            } catch (error) {
+                console.log(error);
+            }
+        }
+
+        // For each entry created in the diff array, generate an HTML table row.
+        let modalItems = diff.map(item => (
+            <tr key={record.id}>
```

Comment:
So React wants your keys to be unique and barks at you in the console when it sees ones that match. I think the keys here will be the same as the keys on line 170. So what I'd recommend here is `<tr key={`modalItems-${record.id}`}>`